### PR TITLE
Clean lib folder

### PR DESCRIPTION
**Pull request by piponazo**
Hi!

Recently I have been playing with some packages depending on zlib and with one of them I have some problems when finding the zlib library. Although the CMake configuration for that project seems to be not correct, I noticed that the zlib recipe is exporting lot of stuff that might not be needed for the package. This is the content of zlib/package/lib on Windows/Static right now:

```
total 539K
-rw-r--r-- 1 Luis 197121 2.3K Jan  3 13:22 adler32.obj
-rw-r--r-- 1 Luis 197121  27K Jan  3 13:22 cl.command.1.tlog
-rw-r--r-- 1 Luis 197121  43K Jan  3 13:22 CL.read.1.tlog
-rw-r--r-- 1 Luis 197121 8.2K Jan  3 13:22 CL.write.1.tlog
-rw-r--r-- 1 Luis 197121 1.9K Jan  3 13:22 compress.obj
-rw-r--r-- 1 Luis 197121  15K Jan  3 13:22 crc32.obj
-rw-r--r-- 1 Luis 197121 1.5K Jan  3 13:22 custombuild.command.1.tlog
-rw-r--r-- 1 Luis 197121 2.3K Jan  3 13:22 custombuild.read.1.tlog
-rw-r--r-- 1 Luis 197121  556 Jan  3 13:22 custombuild.write.1.tlog
-rw-r--r-- 1 Luis 197121  37K Jan  3 13:22 deflate.obj
-rw-r--r-- 1 Luis 197121  692 Jan  3 13:22 gzclose.obj
-rw-r--r-- 1 Luis 197121  13K Jan  3 13:22 gzlib.obj
-rw-r--r-- 1 Luis 197121  14K Jan  3 13:22 gzread.obj
-rw-r--r-- 1 Luis 197121  13K Jan  3 13:22 gzwrite.obj
-rw-r--r-- 1 Luis 197121  13K Jan  3 13:22 infback.obj
-rw-r--r-- 1 Luis 197121 2.7K Jan  3 13:22 inffast.obj
-rw-r--r-- 1 Luis 197121  29K Jan  3 13:22 inflate.obj
-rw-r--r-- 1 Luis 197121 3.9K Jan  3 13:22 inftrees.obj
-rw-r--r-- 1 Luis 197121 5.9K Jan  3 13:22 lib.command.1.tlog
-rw-r--r-- 1 Luis 197121 9.3K Jan  3 13:22 Lib-link.read.1.tlog
-rw-r--r-- 1 Luis 197121 4.8K Jan  3 13:22 Lib-link.write.1.tlog
-rw-r--r-- 1 Luis 197121  20K Jan  3 13:22 trees.obj
-rw-r--r-- 1 Luis 197121 3.2K Jan  3 13:22 uncompr.obj
-rw-r--r-- 1 Luis 197121  253 Jan  3 13:22 zlibstatic.lastbuildstate
-rw-r--r-- 1 Luis 197121 175K Jan  3 13:22 zlibstatic.lib
-rw-r--r-- 1 Luis 197121  33K Jan  3 13:22 zlibstatic.vcxproj
-rw-r--r-- 1 Luis 197121 5.9K Jan  3 13:22 zlibstatic.vcxproj.filters
-rw-r--r-- 1 Luis 197121 2.8K Jan  3 13:22 zutil.obj
```

I think that the only needed file here is the .lib file. 

Note: I am pointing this PR to the release/1.2.9 branch. Let me know if I have to point to other target branch.

### CODE REVIEW

**Comment by piponazo** [comment]
The compilation failed in AppVeyor for the case Windows + MinGW. I will take a look

File: conanfile.py [hunk 1]
```diff
@@ -80,8 +80,7 @@ def package(self):
                 self.copy(pattern="*zlib.lib", dst="lib", src="_build", keep_path=False)
                 self.copy(pattern="*zlib.dll.a", dst="lib", src="_build", keep_path=False)
             else:
-                self.copy(pattern="*zlibstaticd.*", dst="lib", src="_build", keep_path=False)
-                self.copy(pattern="*zlibstatic.*", dst="lib", src="_build", keep_path=False)
+                self.copy(pattern="zlibstatic*.lib", dst="lib", src="_build/lib")
```
Comment (madebr) [inline conanfile.py]: Use only `_build` as `src`. No `_build/lib`.
```suggestion
                self.copy(pattern="zlibstatic*.lib", dst="lib", src="_build")
```

**Comment by SSE4** [comment]
@piponazo I think this is no longer required with latest recipe, WDYT?

**Comment by piponazo** [comment]
Right! I just tested the latest branch and everything looks fine. Thanks!

**Pull request closed without merging**